penn/laundry.py

```diff
@@ -25,37 +29,41 @@ def all_status(self):
         >>> all_laundry = l.all_status()
         """
         r = requests.get(ALL_URL)
-        if r.status_code == 200:
-            parsed = BeautifulSoup(r.text)
-            info_table = parsed.find_all('table')[2]
-
-            hall_dict = dict()
-
-            # This bit of code generates a dict of hallname->hall number by
-            # parsing the link href of each room
-            for link in info_table.find_all('a', class_='buttlink'):
-                def get_hall_no(href):
-                    return int(re.search(r"Halls=(\d+)", href).group(1))
-                hall_dict[link.get_text().strip()] = get_hall_no(link.get('href'))
-
-            # Parse the table into the relevant data
-            data = [list(filter(lambda x: len(x) > 0, [val.get_text().strip() for val in row.find_all('td')])) for row in info_table.find_all('tr')]
-
-            # Remove the header row and all empty rows
-            data_improved = list(filter(lambda x: len(list(x)) > 0, data))[1:]
-
-            # Construct the final JSON
-            laundry_rooms = []
-            for row in data_improved[1:]:
-                room_dict = dict()
-                room_dict['washers_available'] = int(row[1])
-                room_dict['dryers_available'] = int(row[2])
-                room_dict['washers_in_use'] = int(row[3])
-                room_dict['dryers_in_use'] = int(row[4])
-                room_dict['hall_no'] = hall_dict[row[0]]
-                room_dict['name'] = row[0]
-                laundry_rooms.append(room_dict)
-            return laundry_rooms
+        if not r.status_code == 200:
```

Comment:
Now that I think about it, this should not be returning a dictionary, it should raise an exception and be caught by the mobile server. Will use `r.raise_for_status()`.